### DISCUSSION TITLE
[Question]: Fetching Ollama models – confusing info and error

### DISCUSSION THREAD

**Discussion by abdessalaam** [danny-avila/LibreChat#4265] [Q&A]
### What is your question?

I am following the [tutorial]() on adding ollama, and in the sample code there's some information about fetching ollama models: 

```  
# fetching list of models is supported but the `name` field must start
        # with `ollama` (case-insensitive), as it does in this example. 
```

However, the example actually has ``` - name: "Ollama" ``` with the capital "O". 
Then the error in the console says: 
``` 
error: Failed to fetch models from Ollama API. If you are not using Ollama directly, and instead, through some aggregator or reverse proxy that handles fetching via OpenAI spec, ensure the name of the endpoint doesn't start with ollama (case-insensitive). Cannot read properties of undefined (reading 'status')
```

I am quite confused, should it be Ollama or ollama.

### More Details

I have actually tried both and it seems the models are not fetched autimatically either way, and only the ones I list manually are listed. I would appreciate some help with this issue. Thank you!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It should be "Ollama" not "ollama". I hope to make Ollama an official provider to remove the confusion from this

**Reply by abdessalaam** [reply]
Thanks!

**Reply by boydthomson** [reply]
Were you able to get Ollama working with LibreChat?  I've had the same error and am using "Ollama".

**Reply by abdessalaam** [reply]
Yes, it works for me if I list the models I want to use – it doesn't fetch them automatically. My ``` librechat.yaml ``` is: 

``` 
version: 1.1.7
cache: true

endpoints:
  custom:
    - name: "Ollama"
      apiKey: "_"
      # use 'host.docker.internal' instead of localhost if running LibreChat in a docker container
      baseURL: "http://localhost:11434/v1/chat/completions" 
      models:
        default: [
          "llama3.2",
          "llama3.1:8b-instruct-q4_1",
          "qwen2.5",
          "phi3.5",
          "starcoder2:3b"
          ]
        # fetching list of models is supported but the `name` field must start
        # with `ollama` (case-insensitive), as it does in this example.
      fetch: true
      titleConvo: true
      titleModel: "phi3.5"
      summarize: false
      summaryModel: "phi3.5"
      forcePrompt: false
      modelDisplayLabel: "Ollama"

``` 

So I use ``` localhost ``` even though I run LibreChat in Docker (since my ollama is not dockerized).

**Reply by boydthomson** [reply]
Thanks.  I've tried this on two independent servers, with Ollama running as a service, and with Ollama running in a docker container.

Connecting to other APIs (anthropic or openai) works as expected, with RAG.  Thank you!

No matter how I try connecting to Ollama, I get:

2024-10-03 15:33:08 error: [OpenAIClient.chatCompletion] Unhandled error type fetch failed
2024-10-03 15:33:08 error: [handleAbortError] AI response error; aborting request: fetch failed

or

2024-10-03 15:49:40 warn: [OpenAIClient.chatCompletion][stream] API error
2024-10-03 15:49:40 error: [handleAbortError] AI response error; aborting request: Connection error.

Running curl -D - http://localhost:11434/v1/chat/completions -d '{"model": "llama3.2", "messages": [{"role": "user", "content": "hello"}], "stream": false}' gives me a successful (http 200) response.

**Comment by boydthomson** [comment]
Turned out to be an issue with ollama.service not being visible to docker.  Thank you for sharing a known working librechat.yaml

For others with similar problems, I added:
Environment="OLLAMA_HOST=0.0.0.0:11434"
to /etc/systemd/system/ollama.service
